src/Symfony/Component/HttpKernel/Event/ControllerArgumentsEvent.php

```diff
@@ -94,10 +94,16 @@ public function getNamedArguments(): array
     }
 
     /**
-     * @return array<class-string, list<object>>
+     * @template T of class-string|null
+     *
+     * @param T $className
+     *
+     * @return array<class-string, list<object>>|list<object>
+     *
+     * @psalm-return (T is null ? array<class-string, list<object>> : list<object>)
      */
-    public function getAttributes(): array
+    public function getAttributes(string $className = null): array
```

Comment:
Is it not better to make a `getAttribute()` instead?

Reply:
it still returns an array of attributes, as you can have multiple attributes of the same class